src/include/migraphx/op/reshape.hpp

```diff
@@ -125,12 +232,17 @@ struct reshape
             }
         }
 
-        shape s{inputs.front().type(), rdims};
-        if(s.elements() != inputs.front().elements())
+        auto s = reshape_dims(inputs.front(), rdims);
+        if(not s.has_value())
+            MIGRAPHX_THROW("Reshape on axis that is not packed.");
```

Comment:
It doesn't look like this is the only way `s` would not have a value. Wouldn't this also catch the next throw statement with the wrong number of elements?